### ISSUE TITLE
Event on children state is unable to switch to different parent state

### ISSUE THREAD

**Issue by MiChAeLoKGB** [davidkpiano/xstate#93]
## Bug or feature request?
Not really sure...

## Description:
When you want to switch from children state within one parent, to different parent, you gen an error message `Event '[object Object]' on state 'XXX' leads to undefined state 'YYY'.`

## (Bug) Expected result:
I expected it to function similarly to `on: XXX` handlers, which bubble to their parents ->  if the current parent/machine does not have such state, bubble one step up and try to find it again.

## (Bug) Actual result:
`Uncaught Error: Event '[object Object]' on state 'validate' leads to undefined state 'companyData'.`

## Link to reproduction or proof-of-concept:
https://codepen.io/MiChAeLoKGB/pen/YLwXYr

It might just be me not understanding Xstate and statecharts correctly, but I would like to be able to transition to next state **only** from validate children state and **only** on success event.
Right now it looks like I have to add `on: SUCCESS` to the parent, but that would transfer you to next step even without going through validation in some cases...

**Comment by davidkpiano** [comment]
This is related to #52 - and will be handled by adding a unique ID to the state you wish to target, and targeting that with `#stateId`:

```
// ...
start: {
  // ...
  states: {
    // ...
    validate: {
	  on: {
		SUCCESS: { target: '#companyData', cond: () => { ... } }
	  }
	}
  }
},
companyData: {
  id: 'companyData', // or whatever you want
}
```

Your understanding is _mostly correct_ but it's important to keep in mind that targets are relative; that is, `target: 'companyData'` will try to target `'start.companyData'` because that's the sibling of `'start.validate'`.

But if you use IDs, you can avoid that and get the desired behavior. This will be out in 3.2 (which will be released 🔜)

**Comment by davidkpiano** [comment]
Fixed with #52
